### PR TITLE
Add deletion_policy field to Project resource

### PR DESCRIPTION
**Pull request by abd-goog**
Add deletion_policy field to make deletion actions require an explicit intent.
Part of b/330143705

Part of  https://github.com/hashicorp/terraform-provider-google/issues/18775

**Release Note Template for Downstream PRs (will be copied)**
```release-note:enhancement
resourcemanager: added `deletion_policy` field to replace `skip_delete` field in `google_project` resource. Setting `deletion_policy` to `PREVENT` will protect the project against any destroy actions caused by a terraform apply or terraform destroy. Setting `deletion_policy` to `ABANDON` allows the resource to be abandoned rather than deleted and it behaves the same with `skip_delete = true`. Default value is `DELETE`.
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_test.go [hunk 1]
```diff
@@ -350,6 +351,65 @@ func testAccCheckGoogleProjectHasNoLabels(t *testing.T, r, pid string) resource.
...
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"skip_delete", "deletion_policy"},
+			},
+			{
+				Config:      testAccProject_noAllowDestroy(pid, org),
+				Destroy:     true,
+				ExpectError: regexp.MustCompile("deletion_policy"),
+			},
+			{
+				Config: testAccProject(pid, org),
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_test.go:378]: testAccProject is not defined.

Comment (abd-goog) [reply mmv1/third_party/terraform/services/resourcemanager/resource_google_project_test.go:378]: Ahh missed it. Updated.

**Comment by zli82016** [comment]
You need to add `deletion_policy = "NONE"` to all of the places with `google_project` as it is doing in https://github.com/GoogleCloudPlatform/magic-modules/pull/11255/files

**Comment by abd-goog** [comment]
> You need to add `deletion_policy = "NONE"` to all of the places with `google_project` as it is doing in https://github.com/GoogleCloudPlatform/magic-modules/pull/11255/files

Could you please clarify why? The other PR requires it because it was PREVENT by default and hence it broke post test destroy process. In this PR on the other hand its NONE by default and therefore doesn't affect existing flows.

**Comment by zli82016** [comment]
> > You need to add `deletion_policy = "NONE"` to all of the places with `google_project` as it is doing in https://github.com/GoogleCloudPlatform/magic-modules/pull/11255/files
> 
> Could you please clarify why? The other PR requires it because it was PREVENT by default and hence it broke post test destroy process. In this PR on the other hand its NONE by default and therefore doesn't affect existing flows.

Oh, right.

**Review by zli82016 — APPROVED** [review]
LGTM.

**Comment by zli82016** [comment]
Added @slevenick as the secondary review to double check.

**Comment by zli82016** [comment]
I forgot that one more change is needed in the read method. Otherwise, this will be a breaking change.  Please find more information on the PR https://github.com/GoogleCloudPlatform/magic-modules/pull/11330. 

```
	// Explicitly set client-side fields to default values if unset
	if _, ok := d.GetOkExists("deletion_policy"); !ok {
		if err := d.Set("deletion_policy", "NONE"); err != nil {
			return fmt.Errorf("Error setting deletion_policy: %s", err)
		}
	}
```

**Comment by abd-goog** [comment]
@slevenick could you please approve? This is blocking the breaking change on major release branch.

**Review by zli82016 — APPROVED** [review]
LGTM. I tested the projects locally by upgrading the provider with this PR and no perma diffs in the plan after upgrading.

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project.go [hunk 2]
```diff
@@ -73,6 +74,14 @@ func ResourceGoogleProject() *schema.Resource {
 				Computed:    true,
 				Description: `If true, the Terraform resource can be deleted without deleting the Project via the Google API.`,
 			},
+			"deletion_policy": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NONE",
+				Description: `The deletion policy for the Project. Setting PREVENT will protect the project against any destroy actions caused by a terraform apply or terraform destroy. Setting ABANDON allows the resource
+				to be abandoned rather than deleted. Possible values are: "PREVENT", "ABANDON", "NONE"`,
+				ValidateFunc: validation.StringInSlice([]string{"PREVENT", "ABANDON", "NONE"}, false),
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project.go:83]: Can you change `NONE` to `DELETE` to match with other resources?

**Comment by zli82016** [comment]
Added @melinath  as the secondary reviewer.

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project.go (hunk 2 above)
Comment (abd-goog) [reply mmv1/third_party/terraform/services/resourcemanager/resource_google_project.go:83]: You had suggested NONE earlier in https://github.com/GoogleCloudPlatform/magic-modules/pull/11195#issuecomment-2249018282 so could you please confirm it with the 'team' if DELETE is preferred over NONE?  I would appreciate not having a back and forth on this.

Comment (zli82016) [reply mmv1/third_party/terraform/services/resourcemanager/resource_google_project.go:83]: The team prefer `DELETE` over `NONE`. Sorry about that.

Comment (abd-goog) [reply mmv1/third_party/terraform/services/resourcemanager/resource_google_project.go]: Got it. Done.

**Review by melinath — APPROVED** [review]

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**